source/slang/slang-ir.h

```diff
@@ -458,7 +458,7 @@ struct IRInst
 
     void setOperand(UInt index, IRInst* value)
     {
-        getOperands()[index].set(value);
+        getOperands()[index].init(this, value);
```

Comment:
This change seems like something we need to be very careful about.

The current `setOperand()` operates under the assumption that all operands of the instruction have already been initialized (even if it was to a `null` operand value), which means that calling `init(this, ...)` here is working around some piece of code that wasn't following the rules.

I'd rather ass an assert in `set` that the user for the `IRUse` has already been set, and then catch whatever code was violating the rules.

Reply:
Done.

Reply:
I added forced initialization of `IRUse`s in `createInst`, no matter whether the passed in `type` or `args` is null or not.